addons/web/static/src/js/views/search_panel.js

```diff
@@ -58,13 +69,34 @@ function _processSearchPanelNode(node, fields) {
         if (section.type === 'category') {
             section.icon = section.icon || 'fa-folder';
             section.hierarchize = !!pyUtils.py_eval(attrs.hierarchize || '1');
+            if (section.enableCounters) {
+                incompatibilities.category = childNode;
+            }
         } else if (section.type === 'filter') {
             section.domain = attrs.domain || '[]';
+            if (attrs.domain) {
+                incompatibilities.filter = childNode;
+            }
             section.groupBy = attrs.groupby;
             section.icon = section.icon || 'fa-filter';
         }
         sections[sectionId] = section;
     });
+    if (incompatibilities.category && incompatibilities.filter) {
```

Comment:
```js
if (hasFilterWithDomain) {
    const hasSectionWithCounters = Object.values(sections).some(s => s.enableCounters);
    if (hasSectionWithCounters) {
        // if there is a filter with a domain, disable all category counters
        for (const section of Object.values(sections)) {
            if (section.type === "category") {
                section.enableCounters = false;
            }
        }
        // ... and triggers a warning
        console.warn(
            "Warning: categories with counters are incompatible with filters having a domain attribute.",
            "All category counters have been disabled to avoid inconsistencies."
        );       
    }
}
```